classes/source.rb

```diff
@@ -0,0 +1,16 @@
+# Soucr represents a list of sources
+class Source
+  attr_accessor :name, :id
+  attr_reader :items
+
+  def initialize(name, id = nil)
+    @id = id || Random.rand(1..100)
+    @name = name
+    @items = []
+  end
+
+  def add_item(item)
+    @items << item
+    item.add_source(self)
+  end
```

Comment:
- [ ] Secondly, I realized when I check the list of `movies` I get the publish and source details added, but when I tried to check the list of sources, and didn't get any of the added data to display. I believe this is a bug in your project and i would advice you to fix it

Screenshot 👇 

![catalog2](https://github.com/2004-okari/CATALOG/assets/78634841/982eaaff-d338-4abb-bc61-aa2120ba36f6)